File: 92.py
n=0
for i in range(2,10_000_000+1):
    s = str(i)
    while int(s)!=1 and int(s)!=89:
        s = str(sum([int(x)**2 for x in s]))
    if s=='89':
        n+=1
print(n)

File: projecteuler/92.cpp
// g++ 92.cpp -o .exe
// ./.exe

#include <iostream>
using namespace std;

int nextNumber(int x) {
  int sum = 0;
  while (x > 0) {
    int digit = x % 10;
    sum += digit * digit;
    x /= 10;
  }
  return sum;
}

int main() {
  int count = 0;

  for (int i = 2; i <= 10000000; i++) {
    int x = i;

    while (x != 1 && x != 89) {
      x = nextNumber(x);
    }

    if (x == 89) {
      count++;
    }
  }

  cout << count << endl;
  return 0;
}
